src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts

```diff
@@ -189,7 +189,21 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 		}
 
 		//Ensure that the focus stays correct
-		this._grid.getActiveCellNode().focus();
+		if(this._grid.getActiveCellNode()) {
+			this._grid.getActiveCellNode().focus();
+		}
+	}
+
+	// This call is to handle reactive changes in check box UI
+	// This DOES NOT fire UI change Events
+	reactiveCheckboxCheck(row: number, value: boolean) {
+		value ? this._selectedCheckBoxLookup[row] = true : delete this._selectedCheckBoxLookup[row];
+
+		// update row to call formatter
+		this._grid.updateRow(row);
+
+		// ensure that grid reflects the change
+		this._grid.scrollRowIntoView(row);
```

Comment:
have you tested the scenario where you have too many rows to fit within one screen and the affected rows are scattered on different places. I believe in that case, you will see the table to be refreshed multiple times and eventually focus goes back to the original place.

Reply:
Yes. The table doesn't get refreshed multiple times. This example affects 3 rows, but the table doesn't show as refreshing multiple times.
![includeExcludeBigTable](https://user-images.githubusercontent.com/31145923/67729735-79d36180-f9af-11e9-9173-3e9dac257d8c.gif)